### PR TITLE
Say role not found instead of role can not be nil in role add

### LINKED ISSUE #1457: CLI: kontena master users role add master_adminTYPO says "Role can't be nil"
The error message is somewhat confusing if role name has typo in it

### PR DESCRIPTION
**Pull request by kke**
Fixes #1457

### CODE REVIEW

File: server/app/routes/v1/users/user_roles.rb [hunk 1]
```diff
@@ -4,18 +4,23 @@
...
+          response.status = 201
+          render('users/show')
+        else
+          response.status = 422
+          {error: outcome.errors.message}
+        end
       else
         response.status = 422
-        {error: outcome.errors.message}
+        {error: 'Role not found'}
```
Comment (SpComb) [inline server/app/routes/v1/users/user_roles.rb]: This should be something like `{error: {role: "Not Found"}}` to match the Mutations `errors.messages` structure.

File: server/app/routes/v1/users/user_roles.rb [hunk 2]
```diff
@@ -4,18 +4,23 @@
...
-      options = {
-        current_user: current_user,
-        user: @user,
-        role: role
-      }
-      outcome = Users::AddRole.run(options)
-      if outcome.success?
-        response.status = 201
-        render('users/show')
+      if role
```
Comment (jakolehm) [inline server/app/routes/v1/users/user_roles.rb]: Imho this validation should be done inside the mutation?

Comment (kke) [inline server/app/routes/v1/users/user_roles.rb]: Moved the role searching by name inside the mutations.

File: server/app/mutations/users/add_role.rb [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
+
     required do
       model :current_user, class: User
       model :user
-      model :role
+      string :role
+    end
+
+    optional do
+      model :role_instance, class: Role
```
Comment (jakolehm) [inline server/app/mutations/users/add_role.rb]: Do we really need this because `role` is passed as a string?

Comment (jakolehm) [inline server/app/mutations/users/add_role.rb]: @kke ping

Comment (kke) [inline server/app/mutations/users/add_role.rb]: No

**Pull request merged**